### PR TITLE
tests: Fix various issues

### PR DESCRIPTION
**Pull request by thrasher-**
# PR Description

Fixes the following test issues:

1) GateIO 64bit alignment for 32bit systems
2) Intermittent 1/20 chance failure of TestRandomSlippage
3) Skips docker specific tests due to its very relaxed filesystem

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How has this been tested

- [x] go test ./... -race
- [x] golangci-lint run

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by shazbert — APPROVED** [review]
LGTM!

File: exchanges/gateio/gateio.go [hunk 1]
```diff
@@ -173,8 +173,8 @@ var (
 
 // Gateio is the overarching type across this package
 type Gateio struct {
+	Counter common.Counter // Must be first	due to alignment requirements
```
Comment (gloriousCode) [inline exchanges/gateio/gateio.go:176]: This realignment due to the requirement for assignment alignment brings no excitement, only chastisement that my review indictment brought defilement 😭

**Review by gloriousCode — APPROVED** [review]
tACK!

File: exchanges/gateio/gateio.go (hunk 1 above)
Comment (shazbert) [reply exchanges/gateio/gateio.go:176]: 💯 🥇 😗

**Pull request merged**